### PR TITLE
user can delete a favorites list

### PR DESCRIPTION
**Pull request by trigal2012**
## Description

From the favorites list, a user can select a list, once selected, they now also have the option to delete it. 

language files have been updated for all text, the Delete button, the delete confirmation message

button text is also generated from the language files and value is passed as a prop to the button component

other tweaks:
- AsylumConnectButton - add some margin to the base button to remove overlap when there are multiple buttons in the dialog
- DeleteAccountDialog - change the Cancel button to secondary to better distinguish it from the Delete button. - Delete List Dialog and Delete Account Dialog will have a similar style.

## Asana ticket:
https://app.asana.com/0/1200689352502326/1200865997295677

## PR Checklist

- [x] Assign @FJKhan **and** @Alfredo-Moreira as reviewers.
- [x] If your PR is not a hotfix, is it targeted for `dev`? If it is a hotfix, is it targeted for `main`?
- [x] Unit and functional test coverage was added where applicable.
- [x] CI/CD passes for your PR.
- [x] Complex code is well documented with comments.
- [x] Does the original ticket have test instructions? If not add them below
- [x] Pass QA Gate(manual testing)

## How to Test

1. User clicks the Delete button, a dialog opens which will have a confirmation message, the list title and two buttons. If the list had previously been shared an additional message will appear. along the lines of, this list has been shared.....
2. The user can choose Delete or Cancel. 
3. If they choose 'Cancel' the dialog will close, and the user will remain on the view of the  selected list.
4. If they choose "Delete", the list will be deleted from the DB. the dialog will close, once closed the user will return the the Favorites list Home view - The list which had been deleted should no longer appear in the Favorites list.

should validate in Desktop as well as mobile view

### CODE REVIEW

File: src/components/FavoritesListMobile.js [hunk 1]
```diff
@@ -210,6 +211,28 @@ const FavoritesListMobile = ({
...
 				)}
+				{list && isOwner && (
+					<AsylumConnectButton
+						variant="primary"
+						className={classes.spacingTop}
+						onClick={() =>
+							session
+								? handleRequestOpen(
+										'deleteList/' +
+											list._id +
```
Comment (FJKhan) [inline src/components/FavoritesListMobile.js]: would prefer interpolated string here

File: cypress/integration/favorites_list_spec.js [hunk 2]
```diff
@@ -8,7 +8,8 @@
 describe('Favorites List Tests', () => {
 
 
-    let viewports = [Cypress.env('desktop'),Cypress.env('tablet'),Cypress.env('mobile')];
+    // let viewports = [Cypress.env('desktop'),Cypress.env('tablet'),Cypress.env('mobile')];
```
Comment (FJKhan) [inline cypress/integration/favorites_list_spec.js]: @trigal2012 remember to re-enable testing on all viewports

File: src/components/DeleteListDialog.js [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+		display: 'flex',
+		flexDirection: 'column',
+		textAlign: 'center'
+	},
+	marginTop: {
+		marginTop: theme.spacing(1)
+	}
+});
+
+class DeleteListDialog extends React.Component {
```
Comment (FJKhan) [inline src/components/DeleteListDialog.js]: please convert to functional component

File: src/components/DeleteListMobile.js [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+	root: {
+		flexGrow: 1,
+		marginLeft: '2.5em',
+		marginRight: '2.5em',
+		borderBottom: '1px solid ' + theme.palette.common.faintBlack,
+		boxShadow: 'none'
+	}
+});
+
+class DeleteListMobile extends React.Component {
```
Comment (FJKhan) [inline src/components/DeleteListMobile.js]: please convert to functional component

File: src/lang/en-CA.json [hunk 5]
```diff
@@ -59,7 +60,7 @@
 	"app.asylum-connect-catalog": "AsylumConnect Catalog",
 	"app.inactivity-sign-in": "Due to inactivity, please sign in to confirm your identity.",
 	"app.welcome": "Welcome to the Canada AsylumConnect Catalog!",
-	"app.search-services": "Find verified LGBTQ+ and immigrant-friendly services",
+	"app.search-services": "Search for verified LGBTQ+ and immigrant-friendly services near you",
```
Comment (FJKhan) [inline src/lang/en-CA.json]: Please fix the language files with the most recent text

File: src/components/DeleteListMobile.js [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+					<DialogTitle data-test-id="delete-list-title">
+						<FormattedMessage id="favorites.delete.list.dialog.title" />"
+						{this.props.listTitle}"?
+					</DialogTitle>
+					{isShared ? (
+						<Typography type="body1" data-test-id="delete-list-shared">
+							<FormattedMessage id="favorites.delete.list.dialog.shared.message" />
+						</Typography>
+					) : (
+						''
```
Comment (FJKhan) [inline src/components/DeleteListMobile.js]: return null here instead of empty string

File: src/components/FavoritesList.js [hunk 7]
```diff
@@ -279,7 +280,7 @@ const FavoritesList = ({
 													'share/collection/' + list._id + '/' + list.name
 											  )
 											: handleMessageNew(
-													'You must be logged in to share resources'
+													'You must be logged in to share resources.'
```
Comment (FJKhan) [inline src/components/FavoritesList.js]: @trigal2012 can we get this text localised please?

**Review by FJKhan — CHANGES_REQUESTED** [review]

File: cypress/integration/favorites_list_spec.js (hunk 2 above)
Comment (trigal2012) [reply cypress/integration/favorites_list_spec.js]: they are enabled. I think you are looking at an older file. This one shows "Outdated".

File: src/components/DeleteListDialog.js [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+		display: 'flex',
+		flexDirection: 'column',
+		textAlign: 'center'
+	},
+	marginTop: {
+		marginTop: theme.spacing(1)
+	}
+});
+
+function DeleteListDialog(props) {
```
Comment (FJKhan) [inline src/components/DeleteListDialog.js]: this should be `const DeleteListDialog = (props) => {...} ` @trigal2012

File: src/components/DeleteListMobile.js [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+	root: {
+		flexGrow: 1,
+		marginLeft: '2.5em',
+		marginRight: '2.5em',
+		borderBottom: '1px solid ' + theme.palette.common.faintBlack,
+		boxShadow: 'none'
+	}
+});
+
+function DeleteListMobile(props) {
```
Comment (FJKhan) [inline src/components/DeleteListMobile.js]: @trigal2012  same comment as DeleteListDialog

File: src/components/DeleteListMobile.js [hunk 10]
```diff
@@ -0,0 +1,124 @@
...
+				<AsylumConnectButton
+					variant="primary"
+					onClick={confirmDelete}
+					className={classes.spacingTop}
+					testIdName="delete-list-delete-button"
+					children={<FormattedMessage id="action.delete" />}
+				></AsylumConnectButton>
+
+				<AsylumConnectButton
+					variant="secondary"
```
Comment (FJKhan) [inline src/components/DeleteListMobile.js:120]: @trigal2012  you don't specify `children` like this.

this is all you need:

```
<AsylumConnectButton
 	variant="secondary"
 	className={classes.spacingTop}
 	testIdName="delete-list-cancel-button"
 	onClick={handleRequestClose}
 >
       <FormattedMessage id="action.cancel" />
</AsylumConnectButton>
```

File: src/lang/en-CA.json [hunk 11]
```diff
@@ -21,15 +21,17 @@
 	"account.leave-organisation-help": "Before joining a new organization, you must leave your current organization.",
 	"account.leave-organisation": "Leave organization",
 	"action.confirmation": "Confirmation",
-	"action.password-reset-success": "Your password has been reset sucessfully! Please check your email.",
+	"action.password-reset-success": "Your password has been reset successfully! Please check your email.",
 	"action.confirm-password": "Confirm Password",
 	"action.share": "Share",
+	"action.share-question": "Share ",
```
Comment (FJKhan) [inline src/lang/en-CA.json]: @trigal2012 this seems to be a duplication of line 26. why is this necessary?

File: src/lang/en-CA.json [hunk 12]
```diff
@@ -117,20 +119,33 @@
 	"error.invalid-request": "Invalid request",
 	"error.no-location-entered": "Please enter a city or state in the location search box above.",
 	"error.sign-in-to-share-resources": "Oops! You need to be signed in to share resources.",
+	"error.sign-in-to-delete-resources": "¡Oops! You need to be signed in to delete resources.",
```
Comment (FJKhan) [inline src/lang/en-CA.json]: @trigal2012 extra  "¡" at beginning.

File: src/lang/en-CA.json [hunk 13]
```diff
@@ -117,20 +119,33 @@
...
 	"error.invalid-reset-token": "Invalid password reset token",
 	"error.required-field-empty": "Please fill out all fields",
 	"error.email-address-mismatch": "The email addresses you have entered do not match",
 	"error.page-not-found": "Page not found",
 	"error.form-description-length": "Description cannot be longer than 1000 characters",
-	"favorites.create-new-list": "Create New Favourites List",
+	"error.session-expired": "Your session has expired. Please log in again.",
```
Comment (FJKhan) [inline src/lang/en-CA.json]: @trigal2012 there's an "inactivity" item already [here](https://github.com/asylum-connect/asylumconnect-catalog/blob/e7cc13e7136c1a5c782706fcb8cc7707b76a04a9/src/lang/en-US.json#L59)
If we're going to use this, it should be "sign in", not "log in"

File: src/lang/en-CA.json [hunk 14]
```diff
@@ -117,20 +119,33 @@
...
+	"favorites.title": "Favourites",
 	"favorites.view-favorites": "View Your Favourites",
+	"favorites.no-lists": "You haven't created any lists yet.",
+	"favorites.create-new-list": "Create a New Favourites List",
 	"favorites.name-list-help": "Name your list by category, day of the week, or the name of whomever this list is for",
-	"favorites.privacy-disclaimer": "Your favourites lists are only visible to you and anyone you share them with.",
+	"favorites.privacy-disclaimer": "Your favourites lists are only visible to you and anyone you choose to share your lists with.",
```
Comment (FJKhan) [inline src/lang/en-CA.json]: @trigal2012 please revert this change.

**Review by FJKhan — CHANGES_REQUESTED** [review]

File: src/lang/en-CA.json (hunk 11 above)
Comment (trigal2012) [reply src/lang/en-CA.json]: @FJKhan 
So this is here because:
in some instances the word Share is a statement and in others it's a question. In Spanish questions begin with an upside down question mark so was looking for a way to to correctly display that.:
	"action.share": "Compartir",
	"action.share-question": "¿Compartir ",
	
Also, I need for there to be a space after the word 'Share' when it was used as a question.


share_question_english
share_question_spanish

File: src/lang/en-CA.json (hunk 14 above)
Comment (trigal2012) [reply src/lang/en-CA.json]: @FJKhan 
ok, I changed this because of what was in the localization link you sent.
Screen Shot 2021-09-30 at 12 46 01 PM

File: src/lang/en-CA.json (hunk 13 above)
Comment (trigal2012) [reply src/lang/en-CA.json]: @FJKhan 
I was basically just converting the text that was already in the page which is why I did the new one.
Should we be using this instead? 
	"app.inactivity-sign-in": "Due to inactivity, please sign in to confirm your identity.",

Comment (trigal2012) [reply src/lang/en-CA.json]: Screen Shot 2021-09-30 at 1 38 32 PM

Comment (trigal2012) [reply src/lang/en-CA.json]: Also, I can make an "error" component to use in all these places.

File: src/lang/en-CA.json (hunk 11 above)
Comment (FJKhan) [reply src/lang/en-CA.json]: agreed for usage in the Spanish context. though for adding a space, an entirely new element is not necessary. the space can be added in the place where the resource is being used. you can remove the space from the files.

File: src/lang/en-CA.json (hunk 13 above)
Comment (FJKhan) [reply src/lang/en-CA.json]: yes, please use: `"app.inactivity-sign-in": "Due to inactivity, please sign in to confirm your identity.",`.
let's not add to the scope of this pr.

File: src/lang/en-CA.json (hunk 14 above)
Comment (FJKhan) [reply src/lang/en-CA.json:134]: understood. however, the file was already marked as completed.

File: src/lang/es-MX.json [hunk 15]
```diff
@@ -114,23 +116,36 @@
 	"error.password-length": "La contraseña debe tener al menos 8 caracteres.",
 	"error.password-mismatch": "Las contraseñas que ingresaste no coinciden.",
 	"error.user-already-exists": "Esta cuenta de usuario ya existe. Intenta iniciar una sesión.",
-	"error.invalid-request": "Solicitud inválida",
+	"error.invalid-request": "Solicitud no válida.",
```
Comment (FJKhan) [inline src/lang/es-MX.json]:
```suggestion
	"error.invalid-request": "Solicitud inválida.",
```

Comment (trigal2012) [reply src/lang/es-MX.json]: @FJKhan 

Screen Shot 2021-09-30 at 9 04 10 PM

File: src/lang/en-CA.json (hunk 14 above)
Comment (FJKhan) [reply src/lang/en-CA.json]: Understood. the existing language is cleaner though

Comment (trigal2012) [reply src/lang/en-CA.json]: @FJKhan ok, I will revert this and also update the localization link. 
Thanks for the feedback!

File: src/lang/en-CA.json (hunk 13 above)
Comment (trigal2012) [reply src/lang/en-CA.json]: @FJKhan sure thing!

File: src/components/DeleteListMobile.js (hunk 6 above)
Comment (FJKhan) [reply src/components/DeleteListMobile.js]: @trigal2012

**Review by FJKhan — CHANGES_REQUESTED** [review]

**Review by FJKhan — APPROVED** [review]

**Comment by FJKhan** [comment]
@trigal2012 can be merged once the all the tests pass
